### PR TITLE
[DOCS] InputEvent*

### PR DESCRIPTION
**Pull request by Nutriz**
Lot of review and talk I guess. I spent lot of time for this without good result so I need a fresh look from another person. I will squash at the end.

### CODE REVIEW

File: doc/classes/InputEvent.xml [hunk 1]
```diff
@@ -64,27 +65,28 @@
...
 			<description>
-				Returns true if the given action is released (i.e. not pressed). Not relevant for the event types MOUSE_MOTION, SCREEN_DRAG and NONE.
+				Returns [code]true[/code] if the given action is released (i.e. not pressed). Not relevant for the event types MOUSE_MOTION, SCREEN_DRAG and NONE.
 			</description>
 		</method>
 		<method name="is_action_type" qualifiers="const">
 			<return type="bool">
 			</return>
 			<description>
+				Returns [code]true[/code] if current input event has a type, a concret implementation and not only the main [code]InputEvent[/code] class.
```
Comment (Noshyaar) [inline doc/classes/InputEvent.xml]: concret

File: doc/classes/InputEventMouse.xml [hunk 2]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="InputEventMouse" inherits="InputEventWithModifiers" category="Core" version="3.0.alpha.custom_build">
-	<brief_description>
 		Base input event type for mouse events.
 	</brief_description>
+	<brief_description>
```
Comment (Noshyaar) [inline doc/classes/InputEventMouse.xml]: This tag is weird.

File: doc/classes/InputEventMouse.xml [hunk 3]
```diff
@@ -58,10 +59,10 @@
 			Mouse button mask identifier, one of or a bitwise combination of the BUTTON_MASK_* constants in [@Global Scope].
 		</member>
 		<member name="global_position" type="Vector2" setter="set_global_position" getter="get_global_position">
-			Global position of the mouse click.
+			Mouse's position relative to the current [Viewport] when used in [method Control._gui_input], otherwise is at 0,0.
 		</member>
 		<member name="position" type="Vector2" setter="set_position" getter="get_position">
-			Local position of the mouse click.
+			Mouse's local position relative to the [Viewport]. If used in [method Control._gui_input] the position is relative to the current [Control] wich is under the mouse.
```
Comment (Noshyaar) [inline doc/classes/InputEventMouse.xml]: wich

File: doc/classes/InputEventScreenDrag.xml [hunk 4]
```diff
@@ -72,13 +73,13 @@
 			Drag event index in the case of a multi-drag event.
 		</member>
 		<member name="position" type="Vector2" setter="set_position" getter="get_position">
-			Position of the drag event.
+			Drag position.
 		</member>
 		<member name="relative" type="Vector2" setter="set_relative" getter="get_relative">
-			Position of the drag event relative to its start position.
+			Drag position relative to it's start position.
```
Comment (Noshyaar) [inline doc/classes/InputEventScreenDrag.xml]: its

File: doc/classes/InputEventMouseMotion.xml [hunk 5]
```diff
@@ -41,10 +42,12 @@
 	</methods>
 	<members>
 		<member name="relative" type="Vector2" setter="set_relative" getter="get_relative">
-			Position of the mouse pointer relative to the previous mouse position.
+			Mouse's position relative to the previous mouse position.
+			/!\ TO COMPLETE with what is the previous mouse position ???
```
Comment (volzhs) [inline doc/classes/InputEventMouseMotion.xml]: AFAIK, the previous position is mouse position at last frame when mouse moves.

File: doc/classes/InputEvent.xml [hunk 6]
```diff
@@ -4,6 +4,7 @@
 		Generic input event
 	</brief_description>
 	<description>
+		Base class of all sort of input event.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: Not a good sentence.  Also, it's not entirely accurate to call it a "base class" in that the built-in nature of the class hides that from the end user. The original `brief_description` was closer, it is a "generic input event".

File: doc/classes/InputEvent.xml [hunk 7]
```diff
@@ -16,7 +17,7 @@
 			<argument index="0" name="event" type="InputEvent">
 			</argument>
 			<description>
-				Returns true if this input event matches the event passed.
+				Returns [code]true[/code] if this input event matches the event passed.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: Returns [code]true[/code] if this event matches [code]event[event].

File: doc/classes/InputEvent.xml [hunk 8]
```diff
@@ -30,14 +31,14 @@
 			<return type="int">
 			</return>
 			<description>
-				Returns the id of the device that generated the event.
+				Returns the device's id that generated the event.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: This is misleading - the _ID_ doesn't generate the event, the _device_ does.

Also, note get/set descriptions will be replaced by the member var's description.

File: doc/classes/InputEvent.xml [hunk 9]
```diff
@@ -46,7 +47,7 @@
 			<argument index="0" name="action" type="String">
 			</argument>
 			<description>
-				Returns true if this input event matches a pre-defined action, no matter the type.
+				Returns [code]true[/code] if this input event matches a pre-defined action, no matter the type.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: "matches a pre-defined action of any type."

File: doc/classes/InputEvent.xml (hunk 1 above)
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: In other descriptions, you've said "if this event has" but in this one you say "current".

What does "concrete implementation" even mean?

Returns [code]true[/code] if this input event's type is one of the [code]InputEvent[/code] constants.

File: doc/classes/InputEvent.xml [hunk 10]
```diff
@@ -64,27 +65,28 @@
 			<argument index="0" name="action" type="String">
 			</argument>
 			<description>
-				Returns true if the given action is released (i.e. not pressed). Not relevant for the event types MOUSE_MOTION, SCREEN_DRAG and NONE.
+				Returns [code]true[/code] if the given action is released (i.e. not pressed). Not relevant for the event types MOUSE_MOTION, SCREEN_DRAG and NONE.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: Put [code] tags around the constant names.

File: doc/classes/InputEvent.xml [hunk 11]
```diff
@@ -55,7 +56,7 @@
 			<argument index="0" name="action" type="String">
 			</argument>
 			<description>
-				Returns true if the given action is being pressed (and is not an echo event for KEY events). Not relevant for the event types MOUSE_MOTION, SCREEN_DRAG and NONE.
+				Returns [code]true[/code] if the given action is being pressed (and is not an echo event for KEY events). Not relevant for the event types MOUSE_MOTION, SCREEN_DRAG and NONE.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: Put [code] tags around the constant names.

File: doc/classes/InputEvent.xml [hunk 12]
```diff
@@ -64,27 +65,28 @@
...
 			<description>
+				Returns [code]true[/code] if current input event has a type, a concret implementation and not only the main [code]InputEvent[/code] class.
 			</description>
 		</method>
 		<method name="is_echo" qualifiers="const">
 			<return type="bool">
 			</return>
 			<description>
-				Returns true if this input event is an echo event (only for events of type KEY, it will return false for other types).
+				Returns [code]true[/code] if this input event is an echo event (only for events of type KEY, it will return false for other types).
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: If you name the `true` case you don't have to say the others are `false`.

Returns [code]true[/code] if this input event is an echo event (only for events of type [code]KEY[code]).

File: doc/classes/InputEvent.xml [hunk 13]
```diff
@@ -64,27 +65,28 @@
...
 			</description>
 		</method>
 		<method name="is_pressed" qualifiers="const">
 			<return type="bool">
 			</return>
 			<description>
-				Returns true if this input event is pressed. Not relevant for the event types MOUSE_MOTION, SCREEN_DRAG and NONE.
+				Returns [code]true[/code] if this input event is pressed. Not relevant for the event types MOUSE_MOTION, SCREEN_DRAG and NONE.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: Add [code] tags to constant names. 

Missing Oxford comma:
types MOUSE_MOTION, SCREEN_DRAG, or NONE.

File: doc/classes/InputEvent.xml [hunk 14]
```diff
@@ -30,14 +31,14 @@
...
-				Returns the id of the device that generated the event.
+				Returns the device's id that generated the event.
 			</description>
 		</method>
 		<method name="get_id" qualifiers="const">
 			<return type="int">
 			</return>
 			<description>
-				Returns the id of the event.
+				Returns the event's id.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: id -> ID

File: doc/classes/InputEvent.xml [hunk 15]
```diff
@@ -124,6 +126,7 @@
 	</methods>
 	<members>
 		<member name="device" type="int" setter="set_device" getter="get_device">
+			Device's ID of the generated the event.
```
Comment (cbscribe) [inline doc/classes/InputEvent.xml]: The event's device ID.

File: doc/classes/InputEventAction.xml [hunk 16]
```diff
@@ -35,8 +36,10 @@
 	</methods>
 	<members>
 		<member name="action" type="String" setter="set_action" getter="get_action">
+			Action's name. Each action is assigned and retrived by this [String].
 		</member>
 		<member name="pressed" type="bool" setter="set_pressed" getter="is_pressed">
+			[code]true[/code] if the action is pressed. [code]false[/code] if the action is released.
```
Comment (cbscribe) [inline doc/classes/InputEventAction.xml]: See Writing Guidelines - boolean members should begin "If true":

If [code]true[/code] the action's state is pressed.

File: doc/classes/InputEventAction.xml [hunk 17]
```diff
@@ -35,8 +36,10 @@
 	</methods>
 	<members>
 		<member name="action" type="String" setter="set_action" getter="get_action">
+			Action's name. Each action is assigned and retrived by this [String].
```
Comment (cbscribe) [inline doc/classes/InputEventAction.xml]: "The action's name." sounds a little better.

"Actions are accessed via this [String]."

**Review by cbscribe — CHANGES_REQUESTED** [review]
I was only able to review part of it so far.

File: doc/classes/InputEventJoypadButton.xml [hunk 18]
```diff
@@ -4,6 +4,7 @@
 		Input event type for joypad button events.
```
Comment (NathanLovato) [inline doc/classes/InputEventJoypadButton.xml]: Joypads can either refer to game controllers and the 4 directional arrows on a controller. It's also a dated word imo.
`Input event for gamepad buttons.`

File: doc/classes/InputEventJoypadButton.xml [hunk 19]
```diff
@@ -4,6 +4,7 @@
...
 	</brief_description>
 	<description>
+		Contains joypad button informations. See [method Node._input] and http://docs.godotengine.org/en/stable/learning/features/inputs/inputevent.html?highlight=input#inputevent.
```
Comment (NathanLovato) [inline doc/classes/InputEventJoypadButton.xml]: `Input event type for gamepad buttons. For joysticks see [InputEventJoypadMotion].`

File: doc/classes/InputEventJoypadButton.xml [hunk 20]
```diff
@@ -49,13 +50,13 @@
...
-			Joypad button identifier, one of the JOY_BUTTON_* constants in [@Global Scope].
+			Button's identifier, one of the JOY_BUTTON_* constants in [@Global Scope].
 		</member>
 		<member name="pressed" type="bool" setter="set_pressed" getter="is_pressed">
-			Pressed state of the joypad button.
+			[code]true[/code] if the button is pressed. [code]false[/code] if the button is released.
 		</member>
 		<member name="pressure" type="float" setter="set_pressure" getter="get_pressure">
-			Intensity of the button pressure, ranges from 0 to 1.0.
+			Button's pressure intensity, ranges from 0 to 1.0.
```
Comment (NathanLovato) [inline doc/classes/InputEventJoypadButton.xml]: `Represents the pressure the user puts on the button with his finger, if the controller supports it. Ranges from [code]0[/code] to [code]1[/code].`

File: doc/classes/InputEventJoypadButton.xml [hunk 21]
```diff
@@ -49,13 +50,13 @@
 	</methods>
 	<members>
 		<member name="button_index" type="int" setter="set_button_index" getter="get_button_index">
-			Joypad button identifier, one of the JOY_BUTTON_* constants in [@Global Scope].
+			Button's identifier, one of the JOY_BUTTON_* constants in [@Global Scope].
```
Comment (NathanLovato) [inline doc/classes/InputEventJoypadButton.xml]: The button's identifier. One of the [code]JOY_BUTTON_*[/code] constants from [@Global Scope].

File: doc/classes/InputEventJoypadMotion.xml [hunk 22]
```diff
@@ -4,6 +4,7 @@
 		Input event type for joypad motion/axis events.
```
Comment (NathanLovato) [inline doc/classes/InputEventJoypadMotion.xml]: Input event type for gamepad joysticks and other motions. For buttons see [InputEventJoypadMotion].`

File: doc/classes/InputEventJoypadMotion.xml [hunk 23]
```diff
@@ -4,6 +4,7 @@
...
 	</brief_description>
 	<description>
+		Contains joypad motion/axis informations. See [method Node._input] and http://docs.godotengine.org/en/stable/learning/features/inputs/inputevent.html?highlight=input#inputevent.
```
Comment (NathanLovato) [inline doc/classes/InputEventJoypadMotion.xml]: `Stores information about joystick motions. One [code]InputEventJoypadMotion[/code] represents one axis at a time.

File: doc/classes/InputEventJoypadMotion.xml [hunk 24]
```diff
@@ -41,10 +42,10 @@
 	</methods>
 	<members>
 		<member name="axis" type="int" setter="set_axis" getter="get_axis">
-			Joypad axis identifier, one of the JOY_AXIS_* constants in [@Global Scope].
+			Axis identifier, one of the JOY_AXIS_* constants in [@Global Scope].
```
Comment (NathanLovato) [inline doc/classes/InputEventJoypadMotion.xml]: Axis identifier. Use one of the [code]JOY_AXIS_*[/code] constants in [@Global Scope].

File: doc/classes/InputEventJoypadMotion.xml [hunk 25]
```diff
@@ -41,10 +42,10 @@
...
 		</member>
 		<member name="axis_value" type="float" setter="set_axis_value" getter="get_axis_value">
-			Position of the axis, ranging from -1.0 to 1.0. A value of 0 means that the axis is in its neutral position.
+			Axis position, ranging from -1.0 to 1.0. A value of 0 means that the axis is in its neutral position.
```
Comment (NathanLovato) [inline doc/classes/InputEventJoypadMotion.xml]: Current position of the joystick on the given axis. The value ranges from [code]-1.0[/code] to [code]1.0[/code]. A value of [code]0[/code] means the axis is in its resting position.

File: doc/classes/InputEventKey.xml [hunk 26]
```diff
@@ -4,6 +4,7 @@
 		Input event type for keyboard events.
 	</brief_description>
 	<description>
+		Contains keyboard events informations. See [method Node._input] and http://docs.godotengine.org/en/stable/learning/features/inputs/inputevent.html?highlight=input#inputevent.
```
Comment (NathanLovato) [inline doc/classes/InputEventKey.xml]: Stores key presses on the keyboard. Supports key presses, key releases and [member echo] events.

File: doc/classes/InputEventKey.xml [hunk 27]
```diff
@@ -63,16 +64,16 @@
 	</methods>
 	<members>
 		<member name="echo" type="bool" setter="set_echo" getter="is_echo">
-			Echo state of the key, i.e. whether it's a repeat event or not.
+			[code]true[/code] if the same key was already pressed before this event, means the key is held down.
```
Comment (NathanLovato) [inline doc/classes/InputEventKey.xml]: `[code]true[/code] if the key was already pressed before this event. It means the user is holding the key down.`

File: doc/classes/InputEventKey.xml [hunk 28]
```diff
@@ -63,16 +64,16 @@
...
 		</member>
 		<member name="pressed" type="bool" setter="set_pressed" getter="is_pressed">
-			Pressed state of the key.
+			[code]true[/code] if the key is pressed. [code]false[/code] if the key is released.
 		</member>
 		<member name="scancode" type="int" setter="set_scancode" getter="get_scancode">
-			Scancode of the key, one of the KEY_* constants in [@Global Scope].
+			Key's scancode, one of the KEY_* constants in [@Global Scope].
```
Comment (NathanLovato) [inline doc/classes/InputEventKey.xml]: Key's scancode, one of the [code]KEY_*[/code] constants in [@Global Scope].

File: doc/classes/InputEventKey.xml [hunk 29]
```diff
@@ -63,16 +64,16 @@
...
 		</member>
 		<member name="unicode" type="int" setter="set_unicode" getter="get_unicode">
-			Unicode identifier of the key (when relevant).
+			Key's unicode identifier when relevant.
```
Comment (NathanLovato) [inline doc/classes/InputEventKey.xml]: No idea of what "when relevant" means.

**Review by NathanLovato — CHANGES_REQUESTED** [review]
Reviewed a few classes. Can use some improvements and clarifications. Input was one of the most nebulous areas of Godot 2's docs. We've got to be extra careful with this one.

- Remove the links from description tags
- Make sure the descriptions actually describe what the class does. E.g. for ScreenTouch, what does it support? Does it contain info about drag, multitouch, does it support touch order/index ..?
- Thing's xxx is not correct English -> The thing's xxx (can fix it globally with regex, at least emacs sed/regex)

File: doc/classes/InputEventMouse.xml (hunk 3 above)
Comment (Jerome67000) [reply doc/classes/InputEventMouse.xml]: what I must change here ?

File: doc/classes/InputEventAction.xml (hunk 16 above)
Comment (NathanLovato) [reply doc/classes/InputEventAction.xml]: My fault, I had forgotten about this and recommended this sentence to Jerome. Yeah gotta fix it

Comment (Jerome67000) [reply doc/classes/InputEventAction.xml]: Ok right. For the "false" part, I remove ? Would be nice to clarify to have a things saying about releasing...

File: doc/classes/InputEventMouse.xml (hunk 3 above)
Comment (Zireael07) [reply doc/classes/InputEventMouse.xml]: wich to which

**Review by NathanLovato — APPROVED** [review]

**Comment by Jerome67000** [comment]
@NathanLovato this is few modifications to do before merging :
InputEventMouseMotion.speed : can eventually be let without speed unit information ?
InputEventMouseButton.factor : if someone can check on mac and give me what it is
InputEvent.shortcut_match() and InputEvent.xformed_by() : don't understand how this two works
If someone can help me with that it would be nice :)

File: doc/classes/InputEventMouseButton.xml [hunk 30]
```diff
@@ -66,12 +68,13 @@
 			Mouse button identifier, one of the BUTTON_* or BUTTON_WHEEL_* constants in [@Global Scope].
 		</member>
 		<member name="doubleclick" type="bool" setter="set_doubleclick" getter="is_doubleclick">
-			Whether the event is a double-click.
+			If [code]true[/code] the mouse button's state is a double-click. If [code]false[/code] the mouse button's state is released.
 		</member>
 		<member name="factor" type="float" setter="set_factor" getter="get_factor">
+			TO TALK in PR, reduz said : i think it's used for apple touch but i don't remember what it does
```
Comment (bojidar-bg) [inline doc/classes/InputEventMouseButton.xml]: Hm, this "TO TALK in PR" wasn't noticed, https://github.com/godotengine/godot-docs/issues/1490

**Pull request merged**